daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ElementKindParsers.scala

```diff
@@ -119,18 +119,26 @@ class DynamicEscapeSchemeParser(escapeScheme: EscapeSchemeParseEv,
   }
 }
 
-class SequenceCombinatorParser(rd: TermRuntimeData, bodyParser: Parser)
+class SequenceCombinatorParser(rd: TermRuntimeData, childParsers: Seq[Parser])
   extends CombinatorParser(rd) {
   override def nom = "Sequence"
 
   override lazy val runtimeDependencies = Nil
 
-  override lazy val childProcessors = Seq(bodyParser)
+  override lazy val childProcessors = childParsers
+
+  val numChildParsers = childParsers.size
 
   def parse(start: PState): Unit = {
+    var i: Int = 0
+
     start.mpstate.groupIndexStack.push(1L) // one-based indexing
 
-    bodyParser.parse1(start)
+    while (i < numChildParsers) {
+      val parser = childParsers(i)
```

Comment:
childParsers is a Seq, which is implemented in Scala as a LinkedList, which does not have constant time access to random elements. So this loop is essentially O(N<superscript>2</superscript>). Instead of a Seq, we want this to be something with constant time access like an Array, or we want to use things like .head() and .tail() to iterate through the list. An Array will probably be more efficient since I think tail() does alot of work behind the scenes.

Reply:
I think this is wrong. Although the type is Seq, it is actually a Vector. I'd say change type to be Vector[Parser], or maybe Array[Parser]. I think Arrays or more efficient than vectors if all we need is random access.

Reply:
This should be addressed in the most recent commit